### PR TITLE
Docs for confidential client application

### PR DESCRIPTION
**Pull request by sangonzal**

### CODE REVIEW

File: lib/msal-node/README.md [hunk 1]
```diff
@@ -81,8 +77,10 @@ npm install @azure/msal-node
 ##  Usage
 
 ### MSAL basics
+- [Understand difference in between Public Client and Confidential Clients](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-client-applications)
 - [Initialize a Public Client Application](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/initialize-public-client-application.md)
-- [Configuration](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/configuration.md)
+- [Initialize a Confidential Client Application](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/initialize-confidential-client-application.md)
+- [Advanced configuration](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/configuration.md)
```
Comment (pkanher617) [inline lib/msal-node/README.md:83]: Advanced config should probably go in a separate section than basics

Comment (pkanher617) [reply lib/msal-node/README.md:83]: Also nitpick: `Advanced Configuration` with capital 'C'

File: lib/msal-node/docs/initialize-public-client-application.md [hunk 2]
```diff
@@ -40,22 +35,12 @@ const pca = new msal.PublicClientApplication(publicClientConfig);
 
 By default, MSAL is configured with the `common` tenant, which is used for multi-tenant applications and applications allowing personal accounts (not B2C).
 ```javascript
-const msalConfig = {
-    auth: {
-        clientId: 'your_client_id',
-        authority: 'https://login.microsoftonline.com/common/'
-    }
-};
+    authority: 'https://login.microsoftonline.com/common/'
```
Comment (pkanher617) [inline lib/msal-node/docs/initialize-public-client-application.md:38]: Not sure why these changes were made...?

File: lib/msal-node/README.md [hunk 3]
```diff
@@ -151,6 +149,3 @@ Copyright (c) Microsoft Corporation.  All rights reserved. Licensed under the MI
 ## We Value and Adhere to the Microsoft Open Source Code of Conduct
 
 This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
```
Comment (pkanher617) [inline lib/msal-node/README.md:151]: Nitpick: end with newlines in files.

File: lib/msal-node/docs/initialize-public-client-application.md (hunk 2 above)
Comment (sangonzal) [reply lib/msal-node/docs/initialize-public-client-application.md:38]: For consistency with the confidential client doc. Otherwise would have to include all of the other config values, although this sections only speaks about authority

File: lib/msal-node/README.md (hunk 1 above)
Comment (sangonzal) [reply lib/msal-node/README.md:83]: Capitalized configuration

File: lib/msal-node/docs/initialize-public-client-application.md (hunk 2 above)
Comment (pkanher617) [reply lib/msal-node/docs/initialize-public-client-application.md:38]: Got it, read the other doc. Makes sense now.

File: lib/msal-node/README.md (hunk 3 above)
Comment (sangonzal) [reply lib/msal-node/README.md:151]: Added newline

File: lib/msal-node/README.md (hunk 1 above)
Comment (sangonzal) [reply lib/msal-node/README.md:83]: Also got rid of the "Advanced" part, as that was what it was before.

File: lib/msal-common/docs/request.md [hunk 4]
```diff
@@ -6,9 +6,9 @@ Since MSAL Node supports various authorization code grants, there is support for
...
-The application is sent a URL that can be used to generate an `authorization code`. This `URL` can be opened in a browser of choice, where the user can input their credential, and will be redirected back to the `redirectUri` (registered during the [app registration](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-app-registration)) with an `authorization code`. The `authorization code` can now be redeemed for a `token` with the following step.
+The application is sent a URL that can be used to generate an `authorization code`. This `URL` can be opened in a browser of choice, where the user can input their credential, and will be redirected back to the `redirectUri` (registered during the [app registration](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-app-registration)) with an `authorization code`. The `authorization code` can now be redeemed for a `token` with the following step. Not that if authorization code flow is being done for a public client application, [PKCE](https://tools.ietf.org/html/rfc7636) is recommended.
```
Comment (sameerag) [inline lib/msal-common/docs/request.md:9]: typo `note` instead of `not`

File: lib/msal-common/docs/request.md [hunk 5]
```diff
@@ -6,9 +6,9 @@ Since MSAL Node supports various authorization code grants, there is support for
...
+- [acquireTokenByCode()](https://azuread.github.io/microsoft-authentication-library-for-js/ref/msal-node/classes/_src_client_publicclientapplication_.publicclientapplication.html#acquiretokenbycode): This API is the second leg of the `authorization code grant` for MSAL Node. The request constructed here should be of the type [AuthorizationCodeRequest](https://azuread.github.io/microsoft-authentication-library-for-js/ref/msal-common/modules/_src_request_authorizationcoderequest_.html). The application passed the `authorization code` received as a part of the above step and exchanges it for a `token`. Not that if authorization code flow is being done for a public client application, [PKCE](https://tools.ietf.org/html/rfc7636) is recommended.
```
Comment (sameerag) [inline lib/msal-common/docs/request.md:11]: same typo as above
